File: Quiz1/Q2.cpp
#include <iostream>
#include <vector>
#include <queue>
#include <algorithm>

using namespace std;

int max_experience(int n, int k, vector<int>& a, vector<int>& b) {
    int max_exp = 0;
    int current_exp = 0;
    int max_b = 0;

    for (int t = 0; t < min(n, k); ++t) {
        current_exp += a[t];
        max_b = max(max_b, b[t]);
        int remaining = k - (t + 1);
        max_exp = max(max_exp, current_exp + remaining * max_b);
    }

    return max_exp;
}

int main() {
    int n, k;
    cin >> n >> k;
    vector<int> a(n), b(n);
    for (int i = 0; i < n; ++i) cin >> a[i];
    for (int i = 0; i < n; ++i) cin >> b[i];

    cout << max_experience(n, k, a, b) << endl;
    return 0;
}

File: Quiz1/Q2.java
import java.util.Scanner;
import java.util.List;
import java.util.ArrayList;

public class Main {

    public static int maxExperience(int n, int k, List<Integer> a, List<Integer> b) {
        int maxExp = 0;
        int currentExp = 0;
        int maxB = 0;

        for (int t = 0; t < Math.min(n, k); t++) {
            currentExp += a.get(t);
            maxB = Math.max(maxB, b.get(t));
            int remaining = k - (t + 1);
            maxExp = Math.max(maxExp, currentExp + remaining * maxB);
        }

        return maxExp;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        int k = scanner.nextInt();
        List<Integer> a = new ArrayList<>();
        List<Integer> b = new ArrayList<>();

        for (int i = 0; i < n; i++) {
            a.add(scanner.nextInt());
        }
        for (int i = 0; i < n; i++) {
            b.add(scanner.nextInt());
        }

        System.out.println(maxExperience(n, k, a, b));

        scanner.close();
    }
}
